app/lib/features/news/pages/add_news.dart

```diff
@@ -0,0 +1,301 @@
+import 'package:acter/features/news/model/news_slide_model.dart';
+import 'package:acter/features/news/news_utils/news_utils.dart';
+import 'package:acter/features/news/providers/news_post_editor_providers.dart';
+import 'package:acter/features/news/widgets/news_post_editor/select_action_item.dart';
+import 'package:acter/features/news/widgets/news_post_editor/news_slide_options.dart';
+import 'package:atlas_icons/atlas_icons.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_easyloading/flutter_easyloading.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:go_router/go_router.dart';
+
+class AddNews extends ConsumerStatefulWidget {
+  const AddNews({super.key});
+
+  @override
+  ConsumerState<ConsumerStatefulWidget> createState() => _AddNewsState();
+}
+
+class _AddNewsState extends ConsumerState<AddNews> {
+  //General variable declaration
+  final textController = TextEditingController();
+  NewsSlideItem? selectedNewsPost;
+
+  //Build UI
+  @override
+  Widget build(BuildContext context) {
+    selectedNewsPost = ref.watch(newsStateProvider).currentNewsSlide;
+
+    return Scaffold(
+      appBar: appBarUI(context),
+      body: bodyUI(context),
+      floatingActionButton: actionButtonUI(context),
+    );
+  }
+
+  //App Bar
+  AppBar appBarUI(BuildContext context) {
+    return AppBar(
+      leading: IconButton(
+        onPressed: () => context.pop(),
+        icon: const Icon(Atlas.xmark_circle),
+      ),
+      backgroundColor: selectedNewsPost == null
+          ? Colors.transparent
+          : selectedNewsPost?.backgroundColor,
+      actions: [
+        Visibility(
+          visible: selectedNewsPost?.type == NewsSlideType.text,
+          child: IconButton(
+            onPressed: () {
+              ref
+                  .read(newsStateProvider.notifier)
+                  .changeTextSlideBackgroundColor();
+            },
+            icon: const Icon(Atlas.color),
+          ),
+        ),
+      ],
+    );
+  }
+
+  //Action Button
+  Widget actionButtonUI(BuildContext context) {
+    return Visibility(
+      visible: selectedNewsPost != null,
+      child: Padding(
+        padding: const EdgeInsets.only(bottom: 80),
+        child: FloatingActionButton(
+          onPressed: () => selectActionItemDialog(context),
+          child: const Icon(Atlas.plus_circle),
+        ),
+      ),
+    );
+  }
+
+  //Select any widget for action button
+  void selectActionItemDialog(BuildContext context) {
+    showAdaptiveDialog(
+      context: context,
+      builder: (context) {
+        return AlertDialog.adaptive(
+          title: const Text('Add an action widget'),
+          content: SelectActionItem(
+            onSpaceItemSelected: () async {
+              Navigator.of(context, rootNavigator: true).pop();
+              EasyLoading.showInfo('Coming soon');
+              // await ref
```

Comment:
remove this comment too ;)

Reply:
I have clarify reason on above comment.
https://github.com/acterglobal/a3/pull/1347#discussion_r1475658575